### ISSUE TITLE
[webview_flutter] Add Windows support

### ISSUE THREAD

**Issue by Tumbler0809** [flutter/flutter#37597]
I want to implement the function of writing a piece of code to open a web page in android, ios, windows, but this (https://pub.dev/packages/webview_flutter) does not support the windows platform, when can I support it?

**Comment by stuartmorgan** [comment]
The Windows embedding is still a placeholder, the Windows plugin API is not at all finalized, and there's no tooling support for plugins on Windows. On top of that, this particular plugin needs platform view support, which is non-trivial. Given all of that, there's currently no set time-frame for supporting the plugin.

**Comment by Tumbler0809** [comment]
Okay thank you

**Comment by stuartmorgan** [comment]
Blocked on #41723

**Comment by rupamking1** [comment]
Please add WebView Support for Windows

**Comment by stuartmorgan** [comment]
[WebView2](https://docs.microsoft.com/en-us/microsoft-edge/webview2/) is the most likely candidate for implementation here.

**Comment by metaa** [comment]
> [WebView2](https://docs.microsoft.com/en-us/microsoft-edge/webview2/) is the most likely candidate for implementation here.

That's pretty much what I meant, nice.

**Comment by AngusPei** [comment]
> Blocked on #31713

 #31713 Depending on #30726 has been closed, the current thread is already in progress?

**Comment by stuartmorgan** [comment]
Relevant for future work: https://github.com/bshiva1987/flutter_win_webview is a proof-of-concept simple webview using WebView2. It just pops up over other content since there's no `PlatformView` support.

**Comment by clarkezone** [comment]
This was the original prototype which the above fork is based on: https://github.com/clarkezone/flutter_win_webview

**Comment by brunobnb** [comment]
Any updates?

**Comment by stuartmorgan** [comment]
It's still blocked on #31713.

**Comment by wanjm** [comment]
why is there no milestone or project to group such important issues;

**Comment by ZhenFTW** [comment]
I'm still waiting for updates

**Comment by ErliSoares** [comment]
Feature would help a lot

**Comment by digi604** [comment]
this is quite essential

**Comment by Falamarcao** [comment]
Now with Flutter Windows on beta channel. Is there any milestones for webviews?

**Comment by sliechti-ib** [comment]
agree with others, a much needed feature in desktop.

**Comment by wanjm** [comment]
it is very important ;

**Comment by stuartmorgan** [comment]
Please [vote for issues you are interested in](https://github.com/flutter/flutter/wiki/Issue-hygiene#thumbs-up-reactions), which affects prioritization, rather than posting me-too comments, which just spam everyone following the issue.

**Comment by chongbo2013** [comment]
When can I use webview, urgent

**Comment by jpnaidu07** [comment]
The same issue is asked multiple times on different platforms like StackOverflow but there is no answer, this is just basic important functionality to be fixed. When windows desktop website supports WebView.

**Comment by chongbo2013** [comment]
@jpnaidu07  Just add webview, jsbrige, you can realize all functions, including js videoplayer

**Comment by jpnaidu07** [comment]
@chongbo2013 **https://pub.dev/packages/bridge_webview_flutter** doesn't support web and windows platforms only supports ios and android.

**Comment by jing-pei** [comment]
@chongbo2013 我现在的解决方案是直接编译成web，再嵌入到Windows界面内，当前实现还未分配，更新时间遥遥无期

**Comment by jnschulze** [comment]
@chongbo2013 See [webview_windows](https://pub.dev/packages/webview_windows)

**Comment by mimiaopp** [comment]
help

**Comment by dragonDScript** [comment]
Please, this is a must on all platforms! I know that this comment doesn't really apport much, but webviews are REALLY needed.

**Comment by alexandremcp** [comment]
This resource is very important to help developers. Nice!

**Comment by insinfo** [comment]
@jnschulze
>  See webview_windows

good job, it's a start, maybe it would be interesting to fork the webview_flutter and create an abstraction on top that calls webview_windows in windows.

**Comment by roman-petrov** [comment]
> good job, it's a start, maybe it would be interesting to fork the webview_flutter and create an abstraction on top that calls webview_windows in windows.

I also think it's a quite good idea: start with abstraction / proxy over a  third party package and just replace third-party package with official one when its published.

**Comment by stuartmorgan** [comment]
There's no need to fork `webview_flutter`, or to invent a new abstraction, if your goal is to use that implementation temporarily. Someone just needs to make [an unendorsed implementation](https://docs.flutter.dev/development/packages-and-plugins/developing-packages#non-endorsed-federated-plugin) of `webview_flutter_platform_interface`'s abstraction which serves as adapter to `webview_windows`. This is one of the key advantages of the federated model.

**Comment by ismaelReis** [comment]
Something about about linux support?

**Comment by rekire** [comment]
Since the Windows platform is now stable, what is the current status of this ticket? The `webview_windows` widget looks okay. How can I find supported federated plugins of `webview_flutter`? Could theoretically the `webview_windows` widget implement the `webview_flutter_platform_interface`? I did not check the requirements to be honest, I'm just curious.

**Comment by stuartmorgan** [comment]
> Since the Windows platform is now stable, what is the current status of this ticket?

The stable release of Windows does not include platform view support, so nothing has changed here.

> How can I find supported federated plugins of `webview_flutter`?

I'm not sure what "supported" means in this context. If you mean supported by the Flutter project, they would be endorsed, so you wouldn't need to find them. If you are looking for third-party implementations, pub.dev search would be your best option.

> Could theoretically the `webview_windows` widget implement the `webview_flutter_platform_interface`?

Feature requests for third-party plugins should be directed to the issue trackers for those plugins.

**Comment by bahricanyesil** [comment]
Is there any update or solution to show the web view as a widget on a page on desktop platforms?

**Comment by stuartmorgan** [comment]
See https://github.com/flutter/flutter/issues/37597#issuecomment-537155156 and https://github.com/flutter/flutter/issues/37597#issuecomment-754212582

There won't be any updates here until that issue is closed, since it's blocked on that issue.

**Comment by bahricanyesil** [comment]
> See [#37597 (comment)](https://github.com/flutter/flutter/issues/37597#issuecomment-537155156) and [#37597 (comment)](https://github.com/flutter/flutter/issues/37597#issuecomment-754212582)
> 
> There won't be any updates here until that issue is closed, since it's blocked on that issue.

Thanks for the info, I'm looking forward to a solution in the near future!

**Comment by gslender** [comment]
Wondering what update/progress is happening on this issue?

Focus in Flutter seems to be on things like a new rendering engine (why) and yet we still have inconsistent support across desktop compared to mobile. Flutter was supposed to promise a single framework for developing cross platform, but this type of issue (and the underlying lack of PlatformView across Flutter desktop).

As such, is there anything we can do to refocus the effort to start addressing this??

**Comment by gslender** [comment]
Why mark the comment as off-topic if there is no progress and no explanation as to why there is no progress since it was raise 2 1/2 years ago???

Note that it looks like PlatformView is starting to show signs of being unblocked too ??

flutter_view examples exist for both macOS and Windows 

https://github.com/flutter/flutter/tree/master/examples/flutter_view/

**Comment by stuartmorgan** [comment]
> Why mark the comment as off-topic

Because it is; please read https://github.com/flutter/flutter/wiki/Issue-hygiene#when-will-my-bug-be-fixed and related discussion on that page.

> and no explanation as to why there is no progres

My previous comment clearly explains that.

**Comment by microcoder-py** [comment]
@stuartmorgan any updates on windows support?

**Comment by RakshithSarja** [comment]
@stuartmorgan May I know why the above question regarding webview_flutter support for windows  is marked as Off-topic? Does it mean there is no plan to support webview on windows platform or what  is the plan?

**Comment by Hixie** [comment]
@RakshithSarja As it says in the banner at the top of this page, our [commenting guidelines](https://github.com/flutter/flutter/wiki/Issue-hygiene) ask that people avoid asking for updates, because if an issue has an update, we will post it. See our [contributing guidelines](https://github.com/flutter/flutter/blob/master/CONTRIBUTING.md)

**Comment by crystalarcus** [comment]
Any updates ???

**Comment by MarlonJD** [comment]
Any updates about this? MacOS support will land soon on next release of Flutter probably 3.23. I think we need windows support.

**Comment by mehtahardikr** [comment]
is there any chances of windows support ?

**Comment by xqdoo00o** [comment]
I'm pessimistic that this will ever be realized, simply because Windows and Linux lack both a built-in WebView component and a forceful system ABI upgrade policy.

**Comment by anidotnet** [comment]
You can take a look at Papyrus, if it suites your need - https://github.com/dizitart/papyrus
